packages/components/src/site-thumbnail/index.tsx

```diff
@@ -23,27 +23,26 @@ export const SiteThumbnail = ( {
 	backgroundColor,
 	children,
 	alt,
-	mShotsUrl,
+	mShotsUrl = '',
 	size = 'small',
 	mshotsOption = MSHOTS_OPTION,
 }: Props ) => {
-	const maybeImage = useMshotsImg( mShotsUrl ?? '', mshotsOption );
+	const { src, isLoading, imgRef } = useMshotsImg( mShotsUrl, mshotsOption );
 
-	const src = maybeImage?.src || '';
-	const visible = !! src && mShotsUrl;
-	const textColor = backgroundColor && getTextColorFromBackground( backgroundColor );
+	const color = backgroundColor && getTextColorFromBackground( backgroundColor );
 
 	const className = classnames(
 		'site-thumbnail',
-		visible ? 'site-thumbnail-visible' : '',
+		! isLoading ? 'site-thumbnail-visible' : '',
 		`site-thumbnail__size-${ size }`
 	);
 
 	const loader = mShotsUrl ? 'site-thumbnail-loader' : '';
 
 	return (
-		<div className={ className } style={ { backgroundColor, color: textColor } }>
-			{ ! visible ? <div className={ loader }>{ children }</div> : <img src={ src } alt={ alt } /> }
+		<div className={ className } style={ { backgroundColor, color } }>
+			{ isLoading && <div className={ loader }>{ children }</div> }
+			{ src && <img ref={ imgRef } src={ src } alt={ alt } loading="lazy" /> }
```

Comment:
One thing worth noting: use of `loading="lazy"` requires `width=""` and `height=""` attributes.

We can fix this on the main pull request, though.

Reply:
In chrome/brave the lazy loading is working out of the box without specifying the width or height.

I've checked these documents, and I cannot find any mention of width and height attributes.

- https://developer.mozilla.org/en-US/docs/Web/Performance/Lazy_loading
- https://html.spec.whatwg.org/multipage/urls-and-fetching.html#lazy-loading-attributes
- https://web.dev/lazy-loading-images/

Reply:
Hm. Specifying `width="" height=""` prevents layout shift, which I think we'd like to do.

Here's the most recent issue I could find on it: https://github.com/GoogleChrome/web.dev/issues/4273